### ISSUE TITLE
[New option] lines-between-class-members allow empty lines between properties but not member functions

### ISSUE THREAD

**Issue by Shinigami92** [eslint/eslint#13295]
**What rule do you want to change?**

lines-between-class-members

**Does this change cause the rule to produce more or fewer warnings?**

depends

**How will the change be implemented? (New option, new default behavior, etc.)?**

default behavior as is

new additional option

**Please provide some example code that this change will affect:**

```js
// Lint rule turned on, but no necessary blank lines between properties / instance fields so they can grouped together
class ClassWithInstanceField {
  fieldA = 'Field A'
  fieldB = this.fieldA

  fieldC = 42

  instanceMethodA() {
    return 'abc'
  }

  instanceMethodB() {
    return 'def'
  }
}
```

**What does the rule currently do for this code?**

Instance methods should have blank lines but properties **can** be grouped together

**What will the rule do after it's changed?**

No warnings for missing blank lines between properties

**Are you willing to submit a pull request to implement this change?**

Don't know if I'm experienced enought with eslint code-base  
If someone guide me to a file and say what needs to be changed, I could try

**Comment by Shinigami92** [comment]
https://github.com/typescript-eslint/typescript-eslint/issues/977#issuecomment-628082706

**Comment by anikethsaha** [comment]
Thanks for the issue 

class fields are still in [stage 3](https://github.com/tc39/proposal-class-fields) and eslint supports proposals which are in stage 4

**Comment by Shinigami92** [comment]
Oh ok, good to now.  
`typescript-eslint/typescript-eslint` told me I should create this here so it's more a TypeScript kind of problem on my side.

**Comment by anikethsaha** [comment]
IMO, we should wait until the proposal lands or it can be done through a custom rule as I think [babel-eslint](https://github.com/babel/babel-eslint) might support the parsing of fields.

**Comment by mdjermanovic** [comment]
We should absolutely discuss this if and when this syntax reaches stage 4.

In the meantime, the existing [`"exceptAfterSingleLine": true`](https://eslint.org/docs/rules/lines-between-class-members#options) option might be useful.
